### ISSUE TITLE
Running Heketi on Kubernetes cluser with one node with 50gb space

### ISSUE THREAD

**Issue by MiodragL** [heketi/heketi#1255]
We have successfully reloaded this topology. json file with heketi-cli: 

```
    "clusters": [
        {
            "nodes": [[]()
                {
                    "node": {
                        "hostnames": {
                            "manage": [
                                "node1" 
                            ],
                            "storage": [
                                "172.xx.xx.xx" 
                            ]
                        },
                        "zone": 1
                    },
                    "devices": [
                        "/dev/sdb"
                    ]
                }
            ]
        }
    ]
}
```
After running 
`heketi-cli volume create --durability=none --size=12`
The output is : 
**Error: Failed to allocate new volume: No space**

Could it be because we only have only one kubernetes node? 

Thanks

**Comment by phlogistonjohn** [comment]
Because you provided the `--durability=none` option it should work with a single node.
There are some scenarios in which heketi will report an "out of space" error when the size of the device is not the limiting factor. Could you please provide the version of heketi you are running, the output of `heketi-cli topology info`, and a portion of the heketi logs related to the create operation you are performing?

**Comment by phlogistonjohn** [comment]
The issue is most likely due to having neither the File nor Block flag on the cluster set.
```
    File:  false
    Block: false
```

Since you are creating a file volume heketi will look for a free cluster with the file flag enabled. Since you have one cluster without the flag it returns the out of space error.

Please look into the command `heketi-cli cluster setflags` to change these flags.
I'm not sure how you got into this state.

**Comment by MiodragL** [comment]
@phlogistonjohn    I run  `--file=true` and i get the same error

**Comment by MiodragL** [comment]
We' ve  created cluster with heketi 5.0.1 and after that we' ve used heketi 7.0.0, could this be the problem?

**Comment by phlogistonjohn** [comment]
>  I run --file=true and i get the same error

I'm not entirely sure what you mean by this. I'm assuming you ran the setflags command to try and change the flags of the specified cluster. Did you confirm that the flags in `heketi-cli topology info` now reflect the desired settings?
If so, we will need to look into the logs.

> We' ve created cluster with heketi 5.0.1 and after that we' ve used heketi 7.0.0, could this be the problem?

I suspect that this is at least partly to blame. We've seen other cases where older versions did not have the correct flags, and that is one reason the setflags command was added.

**Comment by MiodragL** [comment]
@phlogistonjohn  we run  ` heketi-cli cluster setflags --file=false  f2338b9f1d15d85f390a6718e246940d`
Yes, file flag should be set to true, and block to false?
Can You tell me how we can see heketi logs?

**Comment by phlogistonjohn** [comment]
> we run heketi-cli cluster setflags --file=false f2338b9f1d15d85f390a6718e246940d

You probably want file=true because you're creating file volumes.

> Yes, file flag should be set to true, and block to false?

Sure. You can also have block true on the same volume if you eventually want to create some block volumes.

> Can You tell me how we can see heketi logs?

Depends on how heketi is deployed. If you're running heketi as a pod, a command such as `kubectl logs <heketi-pod-name>` should dump the logs to the console. If you're running a standalone heketi, heketi is probably going to log to something like the systemd journal.

> I' ve tried heketi-cli loglevel get,but it prints nothing.

It should only print the status of the internal log level not the contents of the log.

**Comment by phlogistonjohn** [comment]
@MiodragL was my previous comment helpful? If not could you please retest your scenario with the recently released Heketi v8.0.0? Thanks.
